# Capstone Assignment part 2

Table of contents

1. Introduction
2. Analyze ratio of retaurants to population of Mississauga
3. Use Foursquare APi to located businesses in south Mississauga
4. Shortlist number of restaurants and type
5. Conclusion can a new Italian retaurant be approved by the Mayor?

# Introduction/Business problem

Capstone assignment -Aruna

June-2020

Opportunity Assessment for Capstone project using Foursquare API.

Introduction -

We have been hired by the Mayor of Mississauga to assist in approval for a request to set up Italian restaurant in South Mississauga neighbourhood. We plan to use the Foursquare Api to help the mayor with the decision. we will analyze the are for existing businesses and provide guidance with facts on the area.

Process - 
1. Do we need a restaurant in Mississauga?
   We will analyze the polupation of Mississauga, understand the ratio of people to restaurants.
   Establish if there is a need.

2. Will an italian Restaurant be sucessful? 

We will map the neighbourhood for nearest Coffee shops, Restaurants and entertainment areas. this will help in deciding the zoning. We will create an account in Foursquare api. We will obtain the client ID and Secret after registering with foursquare. we will run a search query to see if there are any italian restaurants in the neighbourhood. this result will then be loaded on a Pandas Dataframe. Using Folium Maps we will visualize this data and explore the venue. we need to ensure there are not amny Italian restaurants in the neighborhood.

the report with the maps will be presented to the Mayors office, so they can make a decision if another italian restaurant is required in the area.


In [1]:

import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_590928f7f599463397dc68aaee279193 = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='TGpEV3YR--A7xLNQBKrl0ImR7rQnMoy7bmn4cpb_e3DG',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3-api.us-geo.objectstorage.service.networklayer.com')

body = client_590928f7f599463397dc68aaee279193.get_object(Bucket='arunasfirstproject-donotdelete-pr-gw4xfiuqojl49x',Key='Number_of_restaurants_per_100,000_population_8112018.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_2 = pd.read_csv(body)
df_data_2.head()


City Figure  Per capita  Date  \
0     Amsterdam   53.4         NaN  2013   
1        Austin  207.2         NaN  2018   
2        Bogotá  418.6         NaN  2017   
3      Brussels  360.4         NaN  2013   
4  Buenos Aires  221.3         NaN  2016   

                                      Source Notes  
0  Trade Association for Horeca and Catering   NaN  
1        Restaurant Guide - Austin Chronicle   NaN  
2               Cámara de Comercio de Bogotá   NaN  
3                         FOD Economie ADSEI   NaN  
4                                     FEHGRA   NaN

Read a CSV file that shows no of restaurants per 100,000 people 
Convert it to a data frame
locate Toronto data.

In [10]:
df = df_data_2[['City','Figure']]

In [11]:
df.tail()

City   Figure
27  Stockholm    108.3
28     Sydney    400.4
29     Taipei    307.6
30      Tokyo  1,099.5
31    Toronto    272.5

For 100,000 people Toronto has 272.5 restaurants.
Mississauga population is 775,000 - No of Food services is 1747
TO keep up to Toronto's average we should have had 775,000 *272.5 /100,000 Restaurants we should have 2111.8 restaurants
Data Shows we have 1747 Food services, hence a gap of 364.

SO yes we can have more restaurants in Mississauga.

*********** End of Part 1 ********

Part 2 ******* BEGINS ********

In [12]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation
import geopy




from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize
import os



print('Folium installed')
print('Libraries imported.')

Folium installed
Libraries imported.


In [15]:
CLIENT_ID = 'WTUDUXM1UNC4YJ2YQV3BJWQXFXD4XFSMFR4Z0BAJWIGLLAEA' # your Foursquare ID
CLIENT_SECRET = 'C4SCID4BHMU1CNT4XUTRIJKW1QMMLTQPPODXGCPYHL5USAGV' # your Foursquare Secret
VERSION = '20200515'
LIMIT = 100
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: WTUDUXM1UNC4YJ2YQV3BJWQXFXD4XFSMFR4Z0BAJWIGLLAEA
CLIENT_SECRET:C4SCID4BHMU1CNT4XUTRIJKW1QMMLTQPPODXGCPYHL5USAGV


In [16]:

address = '1367, Queen Victoria avenue, Mississauga, Ontario'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

43.53498249605366 -79.62144195905238


In [17]:
search_query = 'Restaurant'
radius = 500
print(search_query + ' .... OK!')

Restaurant .... OK!


In [21]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION,  radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=WTUDUXM1UNC4YJ2YQV3BJWQXFXD4XFSMFR4Z0BAJWIGLLAEA&client_secret=C4SCID4BHMU1CNT4XUTRIJKW1QMMLTQPPODXGCPYHL5USAGV&ll=43.53498249605366,-79.62144195905238&v=20200515&radius=500&limit=100'

# DATA used

# Data Used for this Capstone assignment- the approach used was to analyze and Scan all the restaurants and small business in the port credit neighborhood.
1. Foursquare Api was used. i registered for the api, created a client id and client secret
2. based on the location in portcredit the latitude and logitude for the location was found byusing geolocator
3. foursqure api was involed for a rasius of 500 around the portcredit location.
4. results were obtained and we got a Json file of 100 records
5. the response was then mapped to a Pandas data frame
6. this data frame was mapped using folium package.


In [22]:
results = requests.get(url).json()
results
print (results)

{'meta': {'code': 200, 'requestId': '5ed85f55660a9f001b50487c'}, 'response': {'venues': [{'id': '4ca5cc2fd971b1f7e182f9e0', 'name': 'Lorne Park Secondary School', 'location': {'address': '1324 Lorne Park Road', 'crossStreet': 'Truscott', 'lat': 43.53119998536198, 'lng': -79.6240973343538, 'labeledLatLngs': [{'label': 'display', 'lat': 43.53119998536198, 'lng': -79.6240973343538}], 'distance': 472, 'postalCode': 'L5H 3B1', 'cc': 'CA', 'city': 'Mississauga', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['1324 Lorne Park Road (Truscott)', 'Mississauga ON L5H 3B1', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d13d941735', 'name': 'High School', 'pluralName': 'High Schools', 'shortName': 'High School', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/school_', 'suffix': '.png'}, 'primary': True}], 'referralId': 'v-1591238931', 'hasPerk': False}, {'id': '4dd192991838a75196474e5d', 'name': 'Anytime Fitness', 'location': {'address': '1150 Lorne Park Rd', 'lat'

In [23]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

categories  hasPerk  \
0  [{'id': '4bf58dd8d48988d13d941735', 'name': 'H...    False   
1  [{'id': '4bf58dd8d48988d175941735', 'name': 'G...    False   
2  [{'id': '4f04b1572fb6e1c99f3db0bf', 'name': 'S...    False   
3  [{'id': '4bf58dd8d48988d118951735', 'name': 'G...    False   
4  [{'id': '4e39a956bd410d7aed40cbc3', 'name': 'T...    False   

                         id         location.address location.cc  \
0  4ca5cc2fd971b1f7e182f9e0     1324 Lorne Park Road          CA   
1  4dd192991838a75196474e5d       1150 Lorne Park Rd          CA   
2  4f8ef556e4b0d35b8089ad22  2330 South Sheridan Way          CA   
3  4e60449b7d8b854088b48470      1150 Lorne Park Dr.          CA   
4  4f89a94de4b013a980b5dda0                      NaN          CA   

  location.city location.country    location.crossStreet  location.distance  \
0   Mississauga           Canada                Truscott                472   
1   Mississauga           Canada                     NaN                659   
2   Mississauga           Canada                     NaN                199   
3   Mississauga           Canada  at Queen Victoria Ave.                639   
4   Mississauga           Canada                     NaN               1328   

                           location.formattedAddress  \
0  [1324 Lorne Park Road (Truscott), Mississauga ...   
1  [1150 Lorne Park Rd, Mississauga ON L5H 3A7, C...   
2  [2330 South Sheridan Way, Mississauga ON L5J 2...   
3  [1150 Lorne Park Dr. (at Queen Victoria Ave.),...   
4                           [Mississauga ON, Canada]   

                             location.labeledLatLngs  location.lat  \
0  [{'label': 'display', 'lat': 43.53119998536198...     43.531200   
1  [{'label': 'display', 'lat': 43.53051, 'lng': ...     43.530510   
2  [{'label': 'display', 'lat': 43.53325357693356...     43.533254   
3  [{'label': 'display', 'lat': 43.53103989841665...     43.531040   
4  [{'label': 'display', 'lat': 43.54644692756784...     43.546447   

   location.lng location.postalCode location.state  \
0    -79.624097             L5H 3B1             ON   
1    -79.616090             L5H 3A7             ON   
2    -79.622085             L5J 2M4             ON   
3    -79.615677             L5H 3A5             ON   
4    -79.625995                 NaN             ON   

                                 name    referralId  
0         Lorne Park Secondary School  v-1591238931  
1                     Anytime Fitness  v-1591238931  
2                      Public Storage  v-1591238931  
3  Battaglia's Lorne Park Marketplace  v-1591238931  
4         White Oaks Park Tennis Club  v-1591238931

In [25]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name            categories  \
0            Lorne Park Secondary School           High School   
1                        Anytime Fitness  Gym / Fitness Center   
2                         Public Storage      Storage Facility   
3     Battaglia's Lorne Park Marketplace         Grocery Store   
4            White Oaks Park Tennis Club          Tennis Court   
5                           Stoney Pizza           Pizza Place   
6                Lorne park lecture hall           High School   
7                         Pioneer Energy           Gas Station   
8                   Konekt - Mississauga                  None   
9                                Kim gym                  None   
10            The Neziol Insurance Group                  None   
11                    Thompson dentistry                  None   
12             Flintstone Glass & Mirror                  None   
13                  1381 Lakeshore Rd. E                  None   
14                           Supperworks                  None   
15                 White Oaks Montessori                  None   
16                   Ichiban Sushi House      Sushi Restaurant   
17                      East West Karate                  None   
18                  Cuda's Tap and Grill                  None   
19   Tony Morra's Lorne Park Spa & Salon                  None   
20               1st Clarkson Boy Scouts  Other Great Outdoors   
21             Islington Sportsmens Club  Other Great Outdoors   
22                         Woodeden Park                  Park   
23                Abandoned Tennis Court          Tennis Court   
24            Almira Land (1 Star Hotel)            Hotel Pool   
25                         katty's house       Assisted Living   
26                                Jakes!       Assisted Living   
27                          A1 City Auto       Automotive Shop   
28                          Petro-Canada           Gas Station   
29                Sylvan Learning Centre                School   
..                                   ...                   ...   
70                        Whiteoaks Park                  Park   
71                   Clarkson/Lorne Park   Housing Development   
72                         Talka Village   Housing Development   
73                       Mayor Mustachio         Boat or Ferry   
74                           The Kitchen            Food Court   
75  Crafter's Common Crops & Pop Up Shop   Arts & Crafts Store   
76                          tony morra's                  None   
77                          Credit River                 River   
78                First Impression Nails    Salon / Barbershop   
79                Southdown Coin Laundry       Laundry Service   
80      Humber College Broadcasting Hall     College Classroom   
81                           Tecumseh PS     Elementary School   
82                             Le Delice                Bakery   
83      Springbank Greenhouses & Nursery           Flower Shop   
84                  LCBO District Office              Building   
85                             Lakeshore                   Spa   
86                           Metro Plaza              Building   
87                                  CIBC                  Bank   
88                    Lorne Park Variety     Convenience Store   
89                          Day Lily Spa                   Spa   
90                    Daban Translations                Office   
91               St Stephens On The Hill                Church   
92                  Christ Church United                Church   
93                     Rankin Automotive           Gas Station   
94                    Schoolin The 'Hood        Student Center   
95                  Arrow Medical Centre       Doctor's Office   
96                Lorne Park Spa & Salon                   Spa   
97                             Starbucks           Coffee Shop   
98                           Lilys Pizza           Pizza Place   
99        

In [54]:
df.describe

<bound method NDFrame.describe of                                    name                   categories  \
0           Lorne Park Secondary School                  High School   
1                       Anytime Fitness         Gym / Fitness Center   
2                        Public Storage             Storage Facility   
3    Battaglia's Lorne Park Marketplace                Grocery Store   
4           White Oaks Park Tennis Club                 Tennis Court   
5                 White Oaks Montessori                         None   
6               Lorne park lecture hall                  High School   
7                  Lorne Park Jewellers                Jewelry Store   
8                          Stoney Pizza                  Pizza Place   
9                        Pioneer Energy                  Gas Station   
10                 Konekt - Mississauga                         None   
11         Lorne Park Dental Associates             Dentist's Office   
12                  Timjas Tha

In [26]:
df = dataframe_filtered

In [27]:
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans
import sys
!{sys.executable} -m pip install folium


     |████████████████████████████████| 102kB 6.2MB/s ta 0:00:011


In [28]:
import folium

In [29]:
m = folium.Map(location=[latitude, longitude],
                       width=750, height=500)

In [30]:
m

In [31]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) 

# add a red circle marker to represent my house in portcredit
folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='My house',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the businesses as blue circle markers
for lat, lng, label in zip(df.lat, df.lng, df.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

In [32]:
df.groupby('categories').count()

name  address  cc  city  country  crossStreet  \
categories                                                                    
Arts & Crafts Store              1        0   1     0        1            0   
Assisted Living                  2        1   2     1        2            0   
Automotive Shop                  2        2   2     2        2            1   
Bakery                           1        1   1     1        1            0   
Bank                             2        1   2     2        2            1   
Boat or Ferry                    1        0   1     0        1            0   
Breakfast Spot                   1        0   1     1        1            0   
Building                         2        0   2     2        2            0   
Bus Line                         1        0   1     0        1            0   
Chinese Restaurant               1        1   1     1        1            0   
Church                           3        1   3     3        3            0   
Coffee Shop                      1        0   1     1        1            1   
College Classroom                1        1   1     1        1            0   
College Football Field           1        1   1     1        1            1   
Convenience Store                1        1   1     1        1            1   
Cosmetics Shop                   1        1   1     1        1            0   
Design Studio                    1        0   1     0        1            0   
Doctor's Office                  4        2   4     2        4            1   
Elementary School                1        0   1     0        1            0   
Flower Shop                      2        2   2     2        2            2   
Food Court                       1        0   1     0        1            0   
Gas Station                      4        2   4     3        4            1   
Government Building              1        0   1     0        1            0   
Grocery Store                    1        1   1     1        1            1   
Gym / Fitness Center             1        1   1     1        1            0   
High School                      2        1   2     2        2            1   
Hotel Pool                       1        0   1     1        1            0   
Housing Development              2        1   2     1        2            0   
Jewelry Store                    1        1   1     1        1            0   
Laundry Service                  2        2   2     2        2            1   
Library                          1        1   1     1        1            0   
Moving Target                    1        0   1     0        1            0   
Music School                     1        1   1     1        1            0   
Office                           2        1   2     1        2            1   
Other Great Outdoors             2        1   2     2        2            0   
Park                             2        0   2     2        2            0   
Parking                          9        0   9     2        9            0   
Physical Therapist               1        1   1     1        1            1   
Pizza Place                      2        1   2     1        2            0   
Pool                             1        0   1     0        1            0   
Professional & Other Places      1        0   1     1        1            0   
River                            1        0   1     0        1            0   
Salon / Barbershop               3        1   3     2        3            0   
School                           2        2   2     2        2            0   
Skating Rink                     1        1   1     1        1            0   
Spa                              4        1   4     1        4            0   
Storage Facility                 1        1   1     1        1            0   
Student Center                   1        0   1     1        1            0   
Sushi Restaurant                 2        0   2     1        2            0   
T

In [33]:
df.shape

(100, 15)

FILTER out categories that are not food services from the data frame above.

In [38]:

fddf = df[df['categories'].isin(['Pizza Place', 'Sushi Restaurant', 'Chinese Restaurant']) ]

fddf.head()

name          categories               address  cc  \
5              Stoney Pizza         Pizza Place        1200 Vanier Dr  CA   
16      Ichiban Sushi House    Sushi Restaurant                   NaN  CA   
41                James Wok  Chinese Restaurant  1107 Lorne Park Road  CA   
98              Lilys Pizza         Pizza Place                   NaN  CA   
99  Orange Fish Sushi House    Sushi Restaurant                   NaN  CA   

           city country crossStreet  distance  \
5   Mississauga  Canada         NaN       945   
16  Mississauga  Canada         NaN       411   
41  Mississauga  Canada         NaN       826   
98          NaN  Canada         NaN       780   
99          NaN  Canada         NaN       797   

                                     formattedAddress  \
5    [1200 Vanier Dr, Mississauga ON L5H 4C7, Canada]   
16                           [Mississauga ON, Canada]   
41  [1107 Lorne Park Road, Mississauga ON L5H 3A1,...   
98                                           [Canada]   
99                                           [Canada]   

                                       labeledLatLngs        lat        lng  \
5   [{'label': 'display', 'lat': 43.542743, 'lng':...  43.542743 -79.626208   
16  [{'label': 'display', 'lat': 43.53371213499307...  43.533712 -79.626236   
41  [{'label': 'display', 'lat': 43.53197945903734...  43.531979 -79.612070   
98  [{'label': 'display', 'lat': 43.53167568412237...  43.531676 -79.612915   
99  [{'label': 'display', 'lat': 43.53200169277666...  43.532002 -79.612454   

   postalCode state                        id  
5     L5H 4C7    ON  520bedfd11d22b2273e5dfb6  
16        NaN    ON  4c9facd346978cfa142ead7f  
41    L5H 3A1    ON  4bc0fc1e461576b071327b32  
98        NaN   NaN  523b66a811d2c7213038a3b5  
99        NaN   NaN  4da77aaaa86e68edb7c870dc

In [39]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) 

# add a red circle marker to represent my house in portcredit
folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='New Italian Restaurant',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the businesses as blue circle markers
for lat, lng, label in zip(fddf.lat, fddf.lng, fddf.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

In [40]:
fddf.name

5                Stoney Pizza
16        Ichiban Sushi House
41                  James Wok
98                Lilys Pizza
99    Orange Fish Sushi House
Name: name, dtype: object

In [ ]:
********** End of Note book ************